grakn-client/src/main/java/ai/grakn/client/LoaderClient.java

```diff
@@ -0,0 +1,343 @@
+ */
+
+package ai.grakn.client;
+
+import ai.grakn.engine.TaskStatus;
+import ai.grakn.graql.InsertQuery;
+import ai.grakn.util.ErrorMessage;
+import ai.grakn.util.REST;
+import mjson.Json;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+import static ai.grakn.engine.TaskStatus.COMPLETED;
+import static ai.grakn.engine.TaskStatus.FAILED;
+import static ai.grakn.engine.TaskStatus.STOPPED;
+import static ai.grakn.util.REST.Request.KEYSPACE_PARAM;
+import static ai.grakn.util.REST.Request.TASK_LOADER_INSERTS;
+import static ai.grakn.util.REST.Request.TASK_STATUS_PARAMETER;
+import static ai.grakn.util.REST.WebPath.TASKS_URI;
+import static java.util.stream.Collectors.toList;
+
+import static ai.grakn.util.REST.Request.TASK_CLASS_NAME_PARAMETER;
+import static ai.grakn.util.REST.Request.TASK_CREATOR_PARAMETER;
+import static ai.grakn.util.REST.Request.LIMIT_PARAM;
+
+import static ai.grakn.util.REST.Request.TASK_RUN_AT_PARAMETER;
+import static ai.grakn.util.REST.WebPath.TASKS_SCHEDULE_URI;
+import static org.apache.commons.lang.exception.ExceptionUtils.getFullStackTrace;
+
+/**
+ * Client to load qraql queries into Grakn.
+ *
+ * Provides methods to batch insert queries. Optionally can provide a consumer
+ * that will execute when a batch finishes loading.
+ *
+ * @author alexandraorth
+ */
+public class LoaderClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LoaderClient.class);
+
+    private final String POST = "http://%s" + TASKS_SCHEDULE_URI;
+    private final String GET = "http://%s" + TASKS_URI + "/%s";
+
+    private final Consumer<Json> onCompletionOfTask;
+    private final Map<Integer,CompletableFuture> futures;
+    private final Collection<InsertQuery> queries;
+    private final String keyspace;
+    private final String uri;
+
+    private AtomicInteger batchNumber;
+    private Semaphore blocker;
+    private int batchSize;
+    private int blockerSize;
+
+    public LoaderClient(String keyspace, String uri) {
+        this(keyspace, uri, (Json t) -> {});
+    }
+
+    public LoaderClient(String keyspace, String uri, Consumer<Json> onCompletionOfTask){
+        this.uri = uri;
+        this.keyspace = keyspace;
+        this.queries = new HashSet<>();
+        this.futures = new ConcurrentHashMap<>();
+        this.onCompletionOfTask = onCompletionOfTask;
+        this.batchNumber = new AtomicInteger(0);
+
+        setBatchSize(25);
```

Comment:
They are through the Java api (and through migration), but not through the configuration files. The problem I encountered was that, as this is a client, it can't use the `ConfigProperties` object in engine. Do you have an ideas how to go about it?